### PR TITLE
feat: add the ability to generate install functions for plugins

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on adding install configurations to the plugin definitions. 

### Detailed summary
- Added `installConfig` property to `PluginGenConfig` type
- Added `initAbiParams` property to `installConfig` in plugin definitions
- Added `dependencies` property to `installConfig` in plugin definitions
- Updated `useAlchemyProvider` to use new plugin installation method
- Added `installTokenReceiverPlugin` method to `TokenReceiverPlugin`
- Added `installMultiOwnerPlugin` method to `MultiOwnerPlugin`
- Added `installSessionKeyPlugin` method to `SessionKeyPlugin`
- Updated `plugingen` script to generate plugin installation code

> The following files were skipped due to too many changes: `packages/accounts/scripts/plugingen.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/accounts/plugindefs/types.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+import type { Abi, Address, Chain, Hex, parseAbiParameters } from "viem";
 
 export type PluginGenConfig = {
   abi: Abi;
   addresses: Record<number, Address>;
   chain: Chain;
   name: string;
-  // TODO: need to make this configurable to run in CI without requiring this
   rpcUrl?: string;
+  installConfig?: {
```
Comment (moldy530) [inline packages/accounts/plugindefs/types.ts:9]: What do y'all think about making this a requirement actually? I think that would simplify things a lot for consumers of plugins if the developers have to outline the install parameters

**Review by denniswon — APPROVED** [review]
reviewed/synced offline. should add TODOs for follow ups. (uninstallConfigs, etc)

File: packages/accounts/plugindefs/types.ts (hunk 1 above)
Comment (denniswon) [reply packages/accounts/plugindefs/types.ts:9]: I think this should be required.

**Comment by moldy530** [comment]
Will follow up with additional PRs to generate: `uninstall*Plugin` and `is*PluginInstalled` methods.

File: packages/accounts/src/msca/plugins/token-receiver/plugin.ts [hunk 2]
```diff
@@ -11,13 +15,21 @@ import type {
 // ERC6900PluginGen: This file is auto-generated by plugingen
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+type InstallArgs = [];
+export type InstallTokenReceiverPluginParams = {
+  args: Parameters<typeof encodeAbiParameters<InstallArgs>>[1];
+  pluginAddress?: Address;
+  injectedHooks?: InjectedHook[];
+  dependencyOverrides?: FunctionReference[];
```
Comment (denniswon) [inline packages/accounts/src/msca/plugins/token-receiver/plugin.ts:23]: good call to add these overrides optionally here

**Pull request merged**